### PR TITLE
Customized wear and take off menu

### PR DESCRIPTION
**Pull request by OzoneH3**
Inspired by #21770

* Wear items from the ground

auswahl_003

### CODE REVIEW

File: src/game_inventory.cpp [hunk 1]
```diff
@@ -134,18 +134,114 @@ int game::inv_for_id( const itype_id &id, const std::string &title )
...
+                return get_number_string( loc->get_env_resist() );
+            }, _( "ENV" ) );
+        }
+
+    private:
+        std::string get_number_string( int number ) const {
+            return number ? string_format( "<%s>%d</color>", color, number ) : std::string();
+        }
+
+        const std::string &color;
```
Comment (BevapDin) [inline src/game_inventory.cpp]: This seems a bit dangerous. Can you *ensure* that the referred string object stays valid through out the lifetime of this object? Changing this into a value (not reference) would make it safer, and would most likely not cost any performance.

File: src/game_inventory.cpp [hunk 2]
```diff
@@ -134,18 +134,112 @@ int game::inv_for_id( const itype_id &id, const std::string &title )
...
-int game_menus::inv::take_off( player &p )
+class armor_inventory_preset: public inventory_selector_preset
 {
-    return g->inv_for_filter( _( "Take off item" ), [ &p ]( const item & it ) {
-        return p.is_worn( it );
-    }, _( "You don't wear anything." ) );
+    public:
+        armor_inventory_preset( const std::string &color ) {
+
+            static const auto get_number_string = [ &color ]( int number ) {
```
Comment (BevapDin) [inline src/game_inventory.cpp]: `color` is bound to a temporary object via the constructor call in `game_menus::inv::wear`. This *static* lambda is create once and when `wear` is called again, it's still there and has still a *reference* the previously created string object, which is long gone.

Seriously: what's so bad about copying a simple string, especially as it's only a short color name?

And by the way: this lambda should not be static at all. It's capturing the color, but the color can be different on *each* call to this constructor, it's a parameter after all.

**Comment by Coolthulhu** [comment]
Regression: dependent items (power armor components) block takeoff, but are still checked later.
Either drop the dependent item checking altogether (is it used for anything important?) or allow dependent item dropping.

**Comment by OzoneH3** [comment]
Yeah it's never reached and only used for power armor anyway so I removed it.

**Comment by Coolthulhu** [comment]
This will need a more proper cleanup then: the entire dependent items feature has to be purged properly or it will bitrot.

**Comment by OzoneH3** [comment]
Dropping worn items is actually using the dependent feature. Not sure why it's allowed to drop them but not take them off. Not sure what to do now, functionaly wise nothing has changed, it's just like it was before.

**Pull request merged**